src/csharp/Microsoft.Spark/Utils/AssemblyLoader.cs

```diff
@@ -153,7 +166,15 @@ internal static Assembly ResolveAssembly(string assemblyName)
                     }
                 }
 
-                throw new FileNotFoundException($"Assembly '{assemblyName}' file not found '{simpleAsmName}[{string.Join(",", s_extensions)}]' in '{string.Join(",", s_searchPaths)}'");
+                s_logger.LogWarn(
```

Comment:
Why the behavior change?

Reply:
Because we threw an exception in `AssemblyLoader`, the `AssemblyLoadContext.Resolving` event would stop checking any other event handlers that were registered.

https://docs.microsoft.com/en-us/dotnet/api/system.runtime.loader.assemblyloadcontext.resolving?view=netcore-3.1
```If more than one event handler is registered for this event, the event handlers are called in order until an event handler returns a value that isn't null. Subsequent event handlers are ignored.```

If we wanted to keep the the same behavior, then we would constantly need to remove and readd our `AssemblyLoader` event handler with the `AssemblyLoadContext` every time we remove/readd the DependencyProvider.

Reply:
Hmm.  In the batch scenario, if the user sees this warning, it should be proceeded by an exception that will be thrown by the assembly loading class that is using this event handler.

In the interactive scenario the following may happen.  User sees this warning.  The next event handler registered by the DependencyProvider is processed.
- Finds assembly in the probing paths defined in the DependencyProvider.
- Unable to find assembly and an exception is thrown similar to the batch scenario.

Reply:
So, is this a noise (since an exception will be always thrown) or will be this be helpful for further debugging?